engine/table/src/main/java/io/deephaven/engine/table/impl/BlinkTableTools.java

```diff
@@ -183,4 +117,134 @@ public static boolean isBlink(Table table) {
         }
         return Boolean.TRUE.equals(table.getAttribute(Table.BLINK_TABLE_ATTRIBUTE));
     }
+
+    private static class BlinkToAppendOnlyOperation implements QueryTable.MemoizableOperation<QueryTable> {
+        private final QueryTable parent;
+        private final long sizeLimit;
+        private final Object memoKey;
+        private final ColumnSource<?>[] sourceColumns;
+        private final WritableColumnSource<?>[] destColumns;
+
+        private QueryTable resultTable;
+        private BaseTable.ListenerImpl resultListener;
+
+        private BlinkToAppendOnlyOperation(
+                @NotNull final QueryTable parent,
+                final long sizeLimit,
+                @Nullable final Object memoKey) {
+            this.parent = parent;
+            this.sizeLimit = sizeLimit;
+            this.memoKey = memoKey;
+
+            this.sourceColumns = new ColumnSource<?>[parent.numColumns()];
+            this.destColumns = new WritableColumnSource<?>[parent.numColumns()];
+        }
+
+        @Override
+        public String getDescription() {
+            final String sizeLimitStr = sizeLimit == Long.MAX_VALUE ? "Long.MAX_VALUE" : Long.toString(sizeLimit);
+            final String memoKeyStr = memoKey == null ? "NO_MEMOIZATION_KEY" : memoKey.toString();
+            return String.format("BlinkTableTools.blinkToAppendOnly(%s, %s)", sizeLimitStr, memoKeyStr);
+        }
+
+        @Override
+        public String getLogPrefix() {
+            return "BlinkTableTools.blinkToAppendOnly";
+        }
+
+        @Override
+        public MemoizedOperationKey getMemoizedOperationKey() {
+            return memoKey == null ? null : MemoizedOperationKey.blinkToAppendOnly(sizeLimit, memoKey);
+        }
+
+        @Override
+        public Result<QueryTable> initialize(boolean usePrev, long beforeClock) {
+            final Map<String, ? extends ColumnSource<?>> sourceColumns = parent.getColumnSourceMap();
+            final Map<String, WritableColumnSource<?>> resultColumns = new LinkedHashMap<>(sourceColumns.size());
+
+            // note that we do not need to enable prev tracking for an add-only table
+            int colIdx = 0;
+            for (Map.Entry<String, ? extends ColumnSource<?>> sourceEntry : sourceColumns
+                    .entrySet()) {
+                final ColumnSource<?> sourceColumn = sourceEntry.getValue();
+                final WritableColumnSource<?> newColumn = ArrayBackedColumnSource.getMemoryColumnSource(
+                        0, sourceColumn.getType(), sourceColumn.getComponentType());
+                resultColumns.put(sourceEntry.getKey(), newColumn);
+
+                // read and write primitives whenever possible
+                this.sourceColumns[colIdx] = ReinterpretUtils.maybeConvertToPrimitive(sourceColumn);
```

Comment:
I'm sorry I suggested the previous rename, now that I realize this consequence. Maybe I should have suggested parentColumns?